libycp/src/YCPBuiltinString.cc

```diff
@@ -299,28 +299,28 @@ s_substring1 (const YCPString &s, const YCPInteger &i1)
     if (s->isAscii())
     {
 	string ss = s->value ();
-	string::size_type start = i1->value();
+	long long start = i1->value();
```

Comment:
Because they are different types. This whole region deals with subtle signed-unsigned distinctions and string::size_type is an unsigned type.